### DISCUSSION TITLE
Modify generated code by `tonic-build`

### DISCUSSION THREAD

**Discussion by wngr** [hyperium/tonic#1139] [Q&A] [answered]
Hey,

I was wondering whether it's possible to alter some parts of the generated code by `tonic-build`:
I'm generating server and client stubs based of my proto files, but I'd like to enable `tcp_keepalive` on the resulting `Connection` object.
Is that possible? If not, what's the recommended way of achieving that?

Thanks!

**Comment by LucioFranco** [comment, marked as answer]
You don't need to use the `connect` method you can construct the Channel type manually and set the options there.
